libsolidity/codegen/ir/Common.cpp

```diff
@@ -67,6 +68,79 @@ string IRNames::internalDispatch(Arity const& _arity)
 		"_out_" + to_string(_arity.out);
 }
 
+string IRNames::implicitConstructor(ContractDefinition const& _contract)
+{
+	return "constructor_" + _contract.name() + "_" + to_string(_contract.id());
+}
+
+string IRNames::constantValueFunction(VariableDeclaration const& _constant)
+{
+	solAssert(_constant.isConstant(), "");
+	return "constant_" + _constant.name() + "_" + to_string(_constant.id());
+}
+
+string IRNames::localVariable(VariableDeclaration const& _declaration)
+{
+	return "vloc_" + _declaration.name() + '_' + std::to_string(_declaration.id());
+}
+
+string IRNames::localVariable(Expression const& _expression)
+{
+	return "expr_" + to_string(_expression.id());
+}
+
+string IRNames::trySuccessConditionVariable(Expression const& _expression)
+{
+	auto annotation = dynamic_cast<FunctionCallAnnotation const*>(&_expression.annotation());
+	solAssert(annotation, "");
+	solAssert(annotation->tryCall, "Parameter must be a FunctionCall with tryCall-annotation set.");
+
+	return "trySuccessCondition_" + to_string(_expression.id());
+}
+
+string IRNames::tupleComponent(size_t _i)
+{
+	return "component_" + to_string(_i + 1);
+}
+
+string IRNames::zeroValue(Type const& _type, string const& _variableName)
+{
+	return "zero_value_for_type_" + _type.identifier() + _variableName;
+}
+
+string IRNames::getterReturnValues(VariableDeclaration const& _declaration)
+{
+	return suffixedVariableNameList("ret_", 0, _declaration.type()->sizeOnStack());
+}
+
+string IRNames::abiFunctionInputLocals(size_t _inputCount)
+{
+	return suffixedVariableNameList("param_", 0, _inputCount);
+}
+
+string IRNames::abiFunctionOutputLocals(size_t _outputCount)
+{
+	return suffixedVariableNameList("ret_", 0, _outputCount);
+}
+
+string IRNames::internalFunctionInputLocals(Arity const& _arity)
+{
+	return suffixedVariableNameList("in_", 0, _arity.in);
+}
+
+string IRNames::internalFunctionOutputLocals(Arity const& _arity)
+{
+	return suffixedVariableNameList("out_", 0, _arity.out);
+}
```

Comment:
I agree with @chriseth and would suggest to move these five back, so all of those using ``suffixedVariableNameList`` from ``getterReturnValues`` to ``internalFunctionOutputLocals``. All those are local to specific yul functions, aren't they?`I'd say we can just generally consider ``in_``, ``out_``, ``ret_`` and ``param_`` all as things reserved for local use in a function - although we could consider unifying the naming for those as well.

Reply:
sure. I'll move them back.

Reply:
Done.